chainercv/links/model/classification/vgg.py

```diff
@@ -0,0 +1,122 @@
+import collections
+
+import chainer
+import chainer.functions as F
+import chainer.links as L
+
+from chainer.initializers import constant
+from chainer.initializers import normal
+
+
+class VGG16Layers(chainer.Chain):
+
+    def __init__(self, pretrained_model='auto', feature='prob',
+                 initialW=None, initial_bias=None
+                 ):
+        if pretrained_model:
+            # As a sampling process is time-consuming,
+            # we employ a zero initializer for faster computation.
+            if initialW is None:
+                initialW = constant.Zero()
+            if initial_bias is None:
+                initial_bias = constant.Zero()
+        else:
+            # employ default initializers used in the original paper
+            if initialW is None:
+                initialW = normal.Normal(0.01)
+            if initial_bias is None:
+                initial_bias = constant.Zero()
+
+        kwargs = {'initialW': initialW, 'initial_bias': initial_bias}
+
+        super(VGG16Layers, self).__init__()
+
+        with self.init_scope():
+            self.conv1_1 = L.Convolution2D(3, 64, 3, 1, 1, **kwargs)
+            self.conv1_2 = L.Convolution2D(64, 64, 3, 1, 1, **kwargs)
+            self.conv2_1 = L.Convolution2D(64, 128, 3, 1, 1, **kwargs)
+            self.conv2_2 = L.Convolution2D(128, 128, 3, 1, 1, **kwargs)
+            self.conv3_1 = L.Convolution2D(128, 256, 3, 1, 1, **kwargs)
+            self.conv3_2 = L.Convolution2D(256, 256, 3, 1, 1, **kwargs)
+            self.conv3_3 = L.Convolution2D(256, 256, 3, 1, 1, **kwargs)
+            self.conv4_1 = L.Convolution2D(256, 512, 3, 1, 1, **kwargs)
+            self.conv4_2 = L.Convolution2D(512, 512, 3, 1, 1, **kwargs)
+            self.conv4_3 = L.Convolution2D(512, 512, 3, 1, 1, **kwargs)
+            self.conv5_1 = L.Convolution2D(512, 512, 3, 1, 1, **kwargs)
+            self.conv5_2 = L.Convolution2D(512, 512, 3, 1, 1, **kwargs)
+            self.conv5_3 = L.Convolution2D(512, 512, 3, 1, 1, **kwargs)
+            self.fc6 = L.Linear(512 * 7 * 7, 4096, **kwargs)
+            self.fc7 = L.Linear(4096, 4096, **kwargs)
+            self.fc8 = L.Linear(4096, 1000, **kwargs)
+
+        self.functions = collections.OrderedDict([
```

Comment:
I do not recommend this style of network definition. It might cause problems when an instance of this class is copied.
(See https://github.com/chainer/chainer/pull/2810 .)